src/components/dropDown/search/FilterModal.tsx

```diff
@@ -0,0 +1,169 @@
+import { useEffect, useState } from "react";
+import ModalOption from "../../../enum/modalOptionTypes";
+import { useModal } from "../../../hooks/modal/ModalProvider";
+import dropdownStyles from "./FilterModal.module.css";
+import { Button } from "@mui/material";
+import { conceptSet, regionSet, sortSet } from "interface/explore/ExplorePost";
+import { useSearchParams } from "react-router-dom";
+
+function FilterModal() {
+  const { closeModal } = useModal();
+
+  //얘들이 필요한 이유는 버튼이 눌렸을때 해당 버튼이 눌렸다는 상태를 저장하기 위해서.
+  const [savedSort, setSavedSort] = useState<string | null>("");
+  const [savedConcept, setSavedConcept] = useState<string | null>("");
+  const [savedRegion, setSavedRegion] = useState<string | null>("");
+
+  const [query, setQuery] = useSearchParams();
+
+  // Esc 눌렀을때 모달 탈출
+  const handleKeyUp = (event: KeyboardEvent) => {
+    if (event.key === "Escape") {
+      closeModal(ModalOption.FILTER);
+      closeModal(ModalOption.SEARCH);
+    }
+  };
+  // 모달 바깥쪽 눌렀을떄 모달 탈출
+  const handleBackdropClick = (event: React.MouseEvent<HTMLDivElement>) => {
+    if (event.target === event.currentTarget) {
+      closeModal(ModalOption.FILTER);
+      closeModal(ModalOption.SEARCH);
+    }
+  };
+  // 컴포넌트 랜더링시에 한번만 리스너 추가
+  useEffect(() => {
+    document.addEventListener("keyup", handleKeyUp);
+    return () => {
+      document.removeEventListener("keyup", handleKeyUp);
+    };
+    // eslint-disable-next-line
+  }, []);
+
+  const handleClick = (inputData: any, aboutFilter: string) => {
```

Comment:
aboutFilter가 어떤 값인지 좀 명확하지 않아서 filterType? 정도의 이름이나 다른 이름 gpt 한번 돌려서 찾아보시는 것도 좋아보입니다.

그리고 type을 string으로 작성하시는 것보다 아래 방식으로 작성해보시면 좋을 것 같습니다. 
타입은 컴포넌트 밖에 작성해주세요!
```suggestion
type filterType = "sort" | "region" | "concept"
  const handleClick = (inputData: any, aboutFilter: filterType) => {
```